model/configuration.go

```diff
@@ -725,19 +725,41 @@ func (p pipelineTypeUsageMap) setSupported(name string, partials otel.Partials)
 	p.usage(name).supported.Set(partials.PipelineTypes())
 }
 
-type pipelineTypeUsage struct {
+// PipelineTypeUsage contains information about active telemetry on the Configuration
+// and its sources and destinations.
+type PipelineTypeUsage struct {
 	sources      pipelineTypeUsageMap
 	destinations pipelineTypeUsageMap
+
+	// active refers to the top level configuration
+	active otel.PipelineTypeFlags
+}
+
+// ActiveFlagsForDestination returns the PipelineTypeFlags that are active for a destination with given name.
+func (p *PipelineTypeUsage) ActiveFlagsForDestination(name string) otel.PipelineTypeFlags {
+	return p.destinations.usage(name).active
 }
 
-func newPipelineTypeUsage() *pipelineTypeUsage {
-	return &pipelineTypeUsage{
+// ActiveFlags returns the pipeline type flags that are in use by the configuration.
+func (p *PipelineTypeUsage) ActiveFlags() otel.PipelineTypeFlags {
+	return p.active
+}
+
+// setActive sets the top level activeFlags for a pipelineTypeUsage
+func (p *PipelineTypeUsage) setActive(t otel.PipelineType) {
+	if !p.active.IncludesType(t) {
+		p.active.Set(t.Flag())
+	}
```

Comment:
Since this is a bitmask, we don't need to check first.

```suggestion
	p.active.Set(t.Flag())
```